src/main/java/bio/terra/service/tabulardata/azure/StorageTableDao.java

```diff
@@ -0,0 +1,131 @@
+   * @param offset Results will be offset by this much
+   * @param limit Results will be limited to this many
+   * @return The load history for a load tag within the confines of offset and limit.
+   */
+  public List<BulkLoadHistoryModel> getLoadHistory(
+      TableServiceClient tableServiceClient,
+      String datasetName,
+      String loadTag,
+      int offset,
+      int limit) {
+    var tableClient = tableServiceClient.getTableClient(azurfyTableName(datasetName));
+    ListEntitiesOptions options =
+        new ListEntitiesOptions().setFilter(String.format("PartitionKey eq '%s'", loadTag));
+    var pagedEntities = tableClient.listEntities(options, null, null);
+    // This could be more efficient, but would need to implement some sort of index to query on.
+    return pagedEntities.stream()
+        .skip(offset)
+        .limit(limit)
+        .map(StorageTableDao::storageTableEntityBulkFileLoadModel)
+        .collect(Collectors.toList());
+  }
+
+  private static TableEntity bulkFileLoadModelToStorageTableEntity(
+      BulkLoadHistoryModel model, String loadTag, Instant loadTime) {
+    return new TableEntity(loadTag, model.getFileId())
+        .addProperty("load_tag", loadTag)
+        .addProperty("load_time", loadTime)
+        .addProperty("source_name", model.getSourcePath())
+        .addProperty("target_path", model.getTargetPath())
+        .addProperty("state", model.getState().name())
+        .addProperty("file_id", model.getFileId())
+        .addProperty("checksum_crc32c", model.getChecksumCRC())
+        .addProperty("checksum_md5", model.getChecksumMD5())
+        .addProperty("error", model.getError());
+  }
+
+  private static BulkLoadHistoryModel storageTableEntityBulkFileLoadModel(TableEntity tableEntity) {
+    var model =
+        new BulkLoadHistoryModel()
+            .sourcePath(tableEntity.getProperty("source_name").toString())
+            .targetPath(tableEntity.getProperty("target_path").toString())
+            .state(BulkLoadFileState.valueOf(tableEntity.getProperty("state").toString()))
+            .fileId(tableEntity.getProperty("file_id").toString());
+
+    Optional.ofNullable(tableEntity.getProperty("checksum_crc32c"))
+        .ifPresent(o -> model.checksumCRC(o.toString()));
+    Optional.ofNullable(tableEntity.getProperty("checksum_md5"))
+        .ifPresent(o -> model.checksumMD5(o.toString()));
+    Optional.ofNullable(tableEntity.getProperty("error")).ifPresent(o -> model.error(o.toString()));
+    return model;
+  }
+
+  /**
+   * Generate a valid azure storage table name for load history. The resulting name will be the
+   * input truncated to fit {@value #LOAD_HISTORY_TABLE_NAME_SUFFIX} as a suffix in less than 63
+   * characters, with all non-alphanumeric characters stripped out.
+   *
+   * @param tableName The root of the table name
+   * @return A valid azure storage table name with load history suffix.
+   */
+  private static String azurfyTableName(String tableName) {
```

Comment:
If we're making up works I would expect `azurify` not `azurfy`. But I don't think we need to make up a word here. How about
```suggestion
  private static String toAzureTableName(String tableName) {
```
Although it's a private method in a class that's azure specific so it probably doesn't need `azure` in its name either.